app/models/user.rb

```diff
@@ -1,6 +1,43 @@
 class User < ApplicationRecord
-  include Clearance::User
-  has_one :profile, dependent: :destroy
+  scope :visible_for_user, ->(user) { where(visible: true).or(where(id: user&.id)) }
+  scope :members, -> { where('joined_at IS NOT NULL AND left_at is NULL') }
 
-  validates :email, uniqueness: { case_sensitive: false }
+  validates :email,
+    uniqueness: { case_sensitive: false },
+    email_format: true
+  validates :preferred_name, presence: true
+  validates :full_name, presence: true
+
+  before_save do
+    self.email_confirmed = false if email_changed?
+  end
+
+  has_secure_password
+  has_secure_token
+
+  def send_email_confirmation
+    regenerate_token
+    UserMailer.confirm_email(self).deliver_now
+    true
+  end
+
+  def create_membership
+    if !email_confirmed?
+      errors.add :base, 'Please verify your email address'
+    elsif is_member?
+      errors.add :base, 'You are already a member'
+    elsif !update(joined_at: Time.now, left_at: nil)
```

Comment:
I could not think of where this could fail. Is there any case where you think this update might fail?

I would write the method like this:

```ruby
if !email_confirmed?
  errors.add :base, 'Please verify your email address'
  return
end

if is_member?
  errors.add :base, 'You are already a member'
  return
end

update!(joined_at: Time.now, left_at: nil)
```